src/components/ReportUpload/CounterUpload/CounterUpload.js

```diff
@@ -2,46 +2,101 @@ import PropTypes from 'prop-types';
 
-function CounterUpload({ onClose, onFail, onSuccess, open, stripes: { okapi }, udpId }) {
+function CounterUpload({ onClose, onSuccess, open, stripes: { okapi }, udpId }) {
   const [formState, setFormState] = useState({});
   const [showInfoModal, setShowInfoModal] = useState(false);
-  const [infoType, setInfoType] = useState('');
+  const [modalContent, setModalContent] = useState({ title: '', content: null, footer: null });
+
   const intl = useIntl();
 
-  const closeInfoModal = () => {
+  const handleSelectOtherFile = () => {
     setShowInfoModal(false);
-    setInfoType('');
+    formState.form?.reset();
   };
 
-  const openInfoModal = (type) => {
-    setShowInfoModal(true);
-    setInfoType(type);
+  const handleCloseModalsAndReset = () => {
+    onClose();
+    handleSelectOtherFile();
   };
 
-  const showErrorInfo = (err = {}) => {
-    if (err.code === 'REPORTS_ALREADY_PRESENT') {
-      openInfoModal(CounterUpload.overwrite);
-      return;
-    }
+  const errorFooter = () => (
+    <>
+      <Button
+        buttonStyle="primary"
+        onClick={handleSelectOtherFile}
+      >
+        <FormattedMessage id="ui-erm-usage.report.upload.selectAnotherFile" />
+      </Button>
+      <Button onClick={handleCloseModalsAndReset}>
+        <FormattedMessage id="ui-erm-usage.general.close" />
+      </Button>
+    </>
+  );
 
-    closeInfoModal();
+  const labelModalUploadFailed = intl.formatMessage({ id: 'ui-erm-usage.report.upload.modal.label.failed' });
+  const labelModalUpload = intl.formatMessage({ id: 'ui-erm-usage.report.upload.modal.label' });
 
-    let message;
-    if (err.code) {
-      message = intl.formatMessage({ id: `ui-erm-usage.counter.upload.error.${err.code}` });
+  const showErrorInfo = (err = {}) => {
+    // REPORTS_ALREADY_PRESENT
+    if (err.code === 'REPORTS_ALREADY_PRESENT') {
+      setModalContent({
+        title: labelModalUpload,
+        content: (
+          <div>
+            <p><FormattedMessage id="ui-erm-usage.report.upload.reportExists" /></p>
+          </div>
+        ),
+        footer: (
+          <>
+            <Button buttonStyle="primary" id="overwriteYes" onClick={() => setFormState({ ...formState, overwrite: true })}>
+              <FormattedMessage id="ui-erm-usage.general.yes" />
+            </Button>
+            <Button id="overwriteNo" onClick={handleSelectOtherFile}>
+              <FormattedMessage id="ui-erm-usage.general.no" />
+            </Button>
+          </>
+        ),
+      });
+    // All ERRORS with CODES
+    } else if (err.code) {
+      setModalContent({
+        title: labelModalUploadFailed,
+        content: (
+          <div>
+            <p>{intl.formatMessage({ id: `ui-erm-usage.counter.upload.error.${err.code}` })}</p>
+            <details>
+              <summary><b>{intl.formatMessage({ id: 'ui-erm-usage.general.moreInformation' })}</b></summary>
+              <p>{err.details}</p>
+            </details>
```

Comment:
Renders an empty details section if the `details` attribute is not defined. Its not a required attribute in the APIs error response, so there might be cases where its missing.

Reply:
Well, if there is an error code, there should also be details in the response, just like here:
```
{
    "code": "UNSUPPORTED_FILE_FORMAT",
    "message": "The file format is not supported.",
    "details": "java.lang.IllegalArgumentException: Invalid filename"
}
```

If the details are missing it would be a bug and the user should report  this case where only an empty accordion is showing.

Reply:
Well, I don't understand why it is optional. Was that a request?